pkgs/dart_tooling_mcp_server/lib/src/utils/constants.dart

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// A namespace for all the parameter names.
+extension ParameterNames on Never {
```

Comment:
why extend Never rather than just creating a new abstract class?

Reply:
Extensions are a bit more restricted (can't be implemented/extended etc), and don't create a real type (they disappear at runtime entirely) so they are technically cheaper.

Reply:
Note also that static methods on extensions are not actually added to `Never` either, so this has no impact on the `Never` type anywhere it appears.